Dsl2/filter sv blacklist

anoronh4 opened the pull request:
Improvements to filtering:
- Added bed/bedpe files used by pcawg to filter results:
  - svBlacklistBed
  - svBlacklistBedpe
  - svBlacklistFoldbackBedpe
  - svBlacklistTEBedpe
- svBlacklist* files from above are sourced from [pcawg_sv_merge data files](https://bitbucket.org/weischenfeldt/pcawg_sv_merge/src/docker/data/blacklist_files/). Their usage was modeled after the `pcawg6_sv_merge_master.sh` script from the same repo.
- parallelized iAnnotateSV, and defined resource configuration of `.*AnnotateSVBedpe` processes to improve runtime. 
- Fixed header of bedpe, because svtools misreads certain lines with special characters, issue made here: https://github.com/hall-lab/svtools/issues/324 
- The filtering scripts `pair_to_pair_annot.py` and `pair_to_bed_annot.py` converged into and were replaced by `filter_regions_bedpe.py`. The name change makes the intention of the script more obvious.
- added `print` statements and exception handling to python scripts to improve logging/debugging.

anoronh4 committed:
change bedpe filtering script to handle both bedpe and bed regions files, and renamed the python script to better reflect its purpose

anoronh4 committed:
add pcawg blacklist filtering

anoronh4 committed:
added reference definition for pcawg blacklist files for loading them into referenceMap

anoronh4 committed:
fixed filepaths for pcawg SV blacklist files

anoronh4 committed:
bugfix of file name for germline annotation

anoronh4 committed:
reorganized filter_regions_bedpe.py to be more modularized

anoronh4 committed:
cleanup filter_regions_bedpe.py

anoronh4 committed:
add notes

anoronh4 committed:
fix filtering tags for sv annotation

anoronh4 committed:
added pcawg blacklist filtering to Germline SV annotation, and fixed publishing of files

anoronh4 committed:
bugfixes

anoronh4 committed:
bugfix for GermlineAnnotateSVBedpe inputs

anoronh4 committed:
add reference for trasposable element filtering in SV

anoronh4 committed:
added print statements for logging purposes, and made it so iannotatesv processes only 1000 rows at a time

anoronh4 committed:
bugfixes

anoronh4 committed:
fix header of bedpe due to improper handling in svtools

anoronh4 committed:
fix bedpe header

anoronh4 committed:
parallelize iannotatesv

anoronh4 committed:
add resource configuration for *AnnotateSVBedpe processes

anoronh4 committed:
raise resources for *AnnotateSVBedpe

anoronh4 committed:
edited travis test references for filtering SV events

anoronh4 committed:
bugfix

anoronh4 committed:
Merge branch 'dsl2/enhancement/WGS_SV' into dsl2/filterSVBlacklist

anoronh4 committed:
remove duplicated line

gongyixiao commented on conf/references.config:
```diff
@@ -60,6 +60,10 @@ params {
       hlaDat = "${params.reference_base}/hla/hla.dat"
       neoantigenCDNA = "${params.reference_base}/neoantigen/Homo_sapiens.GRCh37.75.cdna.all.fa.gz"
       neoantigenCDS = "${params.reference_base}/neoantigen/Homo_sapiens.GRCh37.75.cds.all.fa.gz"
+      svBlacklistBed   = "${params.genome_base}/sv_calling/pcawg6_blacklist.slop.bed.gz"
+      svBlacklistBedpe = "${params.genome_base}/sv_calling/pcawg6_blacklist.slop.trunc.bedpe.gz"
+      svBlacklistFoldbackBedpe = "${params.genome_base}/sv_calling/pcawg6_blacklist_foldback_artefacts.slop.trunc.bedpe.gz"
+      svBlacklistTEBedpe       = "${params.genome_base}/sv_calling/pcawg6_blacklist_TE_pseudogene.bedpe.gz"
```
Is it possible to add the source of these data so that we could trace back and update them in the future if necessary.

anoronh4 replied:
i have added a note within the `references.config` file and also updated the PR comment to show where the files come from. We should definitely add the source to our docs as well once we are back to hosting that.

gongyixiao commented on containers/iannotatesv/run_iannotatesv.py:
```diff
@@ -1,51 +1,68 @@
 	parser.add_argument('--bedpe', help = 'bedpe file', required = True)
 	parser.add_argument('--genome', default = "hg19", help = 'hg19/hg38/hg18')
+	parser.add_argument('--threads', type=int, default = 4, help = 'number of threads for parallelization')
+	parser.add_argument('--chunk', type=int, dest = "chunk_size", default = 500, help = 'how many records to process with a single thread')
```
Can `--chunk` be calculated as an optimized/reasonable value based on number of record in the bedpe file and number of threads was given? I guess it also related to memory as well. Maybe memory limit will also get involved here. So maybe a preset formula can be defined here to cover the most basic needs? 
I'm thinking this might be a bit difficult for user to figure out a best number here with limited knowledge.

anoronh4 replied:
will remove chunk param.

gongyixiao commented:
Suggestion: You might want to put signatures on codes which written by you @anoronh4 . Up to you.

Why am I seeing both `filter_regions_bedpe.py ` and `pair_to_pair_annot.py` in the container folder together and they are not the same? And what was the file `pair_to_bed_annot.py` which is deleted here?

anoronh4 committed:
remove chunk parameter from run_iannotatesv.py because finetuning it does not meaningfully impact run time

anoronh4 committed:
Merge branch 'dsl2/filterSVBlacklist' of github.com:mskcc/tempo into dsl2/filterSVBlacklist

anoronh4 commented:
> Why am I seeing both `filter_regions_bedpe.py ` and `pair_to_pair_annot.py` in the container folder together and they are not the same? And what was the file `pair_to_bed_annot.py` which is deleted here?

`pair_to_pair_annot.py` and `pair_to_bed_annot.py` were supposed to have converged into and been replaced by `filter_regions_bedpe.py`. I forgot to remove `pair_to_pair_annot.py`. it's now removed and i changed the PR summary to better describe the changes.

anoronh4 committed:
add note about sources for pcawg blacklist files

anoronh4 committed:
add description and authorship info for each python file
